src/main/java/org/jabref/gui/fieldeditors/optioneditors/LangidEditorViewModel.java

```diff
@@ -0,0 +1,57 @@
+package org.jabref.gui.fieldeditors.optioneditors;
+
+import java.util.Arrays;
+import java.util.Collection;
+
+import javax.swing.undo.UndoManager;
+
+import javafx.util.StringConverter;
+
+import org.jabref.gui.autocompleter.SuggestionProvider;
+import org.jabref.logic.integrity.FieldCheckers;
+import org.jabref.model.database.BibDatabaseMode;
+import org.jabref.model.entry.Langid;
+import org.jabref.model.entry.field.Field;
+import org.jabref.model.strings.StringUtil;
+
+public class LangidEditorViewModel extends OptionEditorViewModel<Langid> {
+    private BibDatabaseMode databaseMode;
+
+    public LangidEditorViewModel(Field field, SuggestionProvider<?> suggestionProvider, BibDatabaseMode databaseMode, FieldCheckers fieldCheckers, UndoManager undoManager) {
+        super(field, suggestionProvider, fieldCheckers, undoManager);
+        this.databaseMode = databaseMode;
+    }
+
+    @Override
+    public StringConverter<Langid> getStringConverter() {
+        return new StringConverter<>() {
+            @Override
+            public String toString(Langid object) {
+                if (object == null) {
+                    return null;
+                } else {
+                    return object.getLangid();  // Langid used as both display and value
+                }
+            }
+
+            @Override
+            public Langid fromString(String string) {
```

Comment:
Return `Optional<Langid>`  instead of `null`.

Reply:
I was under the impression that it had to return `Langid` because the superclass `OptionEditorViewModel` requires `StringConverter<T>` to return <T> not Optional<T>?

Reply:
Yes, null is correct here.